diff --git a/distribution/core/constants.js b/distribution/core/constants.js
index 0b884fe769a7955c85e9561c060159e6e8ccc148..4be13f3855dca8db6519c651dbd48b7c48dd3717 100644
--- a/distribution/core/constants.js
+++ b/distribution/core/constants.js
@@ -2,7 +2,8 @@ export const supportsRequestStreams = (() => {
     let duplexAccessed = false;
     let hasContentType = false;
     const supportsReadableStream = typeof globalThis.ReadableStream === 'function';
-    if (supportsReadableStream) {
+    const supportsRequest = typeof globalThis.Request === 'function';
+    if (supportsReadableStream && supportsRequest) {
         hasContentType = new globalThis.Request('https://a.com', {
             body: new globalThis.ReadableStream(),
             method: 'POST',